### PR TITLE
Fix bug #255 by defining a state machine for KWS activation

### PR DESCRIPTION
**Pull request by dargilco**
## Purpose

* Fix bug #255 - do not turn on KWS after TTS playback is done, if the user has previously selected to stop TTS
* This state machine will also help us easily implement optional barge-in when a configuration settings in the JSON indicates that barge-in needs to be supported (if the device has good echo cancellation). This is logged as feature request #262.
If the process I found out that if we are loading a wrong binary files as the KWS table, no error is generated. Logged as bug #264.

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Tested by selecting different sequences of keyboard options 1-3 to make sure everything works as expected. Also tested without keyword table.

### CODE REVIEW

File: clients/cpp-console/src/common/mainAudio.cpp [hunk 1]
```diff
@@ -66,26 +84,54 @@ int main(int argc, char** argv)
...
     IAudioPlayer* player;
     shared_ptr<AgentConfiguration> agentConfig;
     shared_ptr<DialogServiceConnector> dialogServiceConnector;
     
-    auto startKwsIfApplicable = [&]()
+    auto StartKws = [&]()
     {
-        log_t("startKWS called");
-        if (keywordListeningEnabled)
+        log_t("startKwsIfApplicable called");
```
Comment (chschrae) [inline clients/cpp-console/src/common/mainAudio.cpp:96]: looks like you have the wrong function name here

File: clients/cpp-console/src/common/mainAudio.cpp [hunk 2]
```diff
@@ -66,26 +84,54 @@ int main(int argc, char** argv)
...
+            keywordActivationState = KeywordActivationState::Listening;
             log_t("KWS initialized");
         }
-        else {
-            log_t("no model file specified. Cannot start keyword listening");
+    };
+
+    auto PauseKws = [&]()
+    {
+        log_t("PauseKws called");
```
Comment (chschrae) [inline clients/cpp-console/src/common/mainAudio.cpp:111]: on these logs should we print out what the state was when being called?

File: clients/cpp-console/src/common/mainAudio.cpp (hunk 1 above)
Comment (dargilco) [reply clients/cpp-console/src/common/mainAudio.cpp:96]: Good catch. Fixing now.

File: clients/cpp-console/src/common/mainAudio.cpp (hunk 2 above)
Comment (dargilco) [reply clients/cpp-console/src/common/mainAudio.cpp]: Added printout of the KWS state on entry and exit of the functions. Thanks!

**Pull request merged**